### ISSUE TITLE
[Question] Artifact Bundle methods

### ISSUE THREAD

**Issue by marikrg** [HewlettPackard/oneview-sdk-ruby#195]
### Scenario/Intent

I have some questions regarding these three endpoints:

**1. GET/rest/artifact-bundles/backups/archive/{id}**

 I'm able to create a backup bundle calling the method `self.create_backup_from_file!`
 How can I restore this backup bundle?
I suppose there is a method to perform this operation.

**2. PUT/rest/artifact-bundles/backups/{id}**

I can call `self.get_backups(client)` to get a list of backups, and `self.create_backup` to create a backup.
But I couldn't find a way to extract this backup bundle.

**3. PUT/rest/artifact-bundles/{id}**

Looks like the description in the `extract(force = true)` method documentation doesn't match the operation being performed by the method.
Shouldn't be
 ```ruby
 # Extracts the artifact bundle and creates the artifacts on the appliance. 
 ```
instead of
 ```ruby
 # Extracts the existing backup bundle on the appliance and creates all the artifacts.
 #   If there are any artifacts existing, they will be removed before the extract operation.
 ```
?

### Environment Details
 - **OneView SDK Version:** current master repo
 - **OneView Appliance Version:** Synergy + Image Streamer
 - **OneView Client API Version:** 300
 - **Ruby Version:** Any
 - **Platform:** Any

### Steps to Reproduce
Does not apply.

### Expected Result
Does not apply.

### Actual Result
Does not apply.

**Comment by ricardogpsf** [comment]
1-  The backup bundle, when uploaded, already is extracted (in documentation has this description).
2 - The extract method (for already backup uploaded) is not necessary if user use only the Ruby SDK, because when backup is did, the extract of artifacts already occurs). Anyway, I will submit the method extract backup.
3 - The description is wrong. I will change it.

Thanks a lot for that observation. A good question!!

**Comment by marikrg** [comment]
**1. GET/rest/artifact-bundles/backups/archive/{id}**
Yes, in fact it is already extracted.
Please let me rephrase my question.
How can I download the backup archive from the Artifact Bundle?

**2. PUT/rest/artifact-bundles/backups/{id}**
This operation doesn't extract the artifact bundle by itself. It extracts the backup from the Deployment Group, and removes all the existent artifacts associated with that group (as plan scripts, build plans, deployment plans, and  golden images resources). WARN: If there are any artifacts existing, they will be removed before the extract operation.

**Comment by ricardogpsf** [comment]
The download_backup method is used for GET/rest/artifact-bundles/backups/archive/{id}

**Comment by marikrg** [comment]
It still not clear for me.
I think that the archive download is missing.
I can only download a backup, not a archive backup.

**Comment by ricardogpsf** [comment]
The method `download_backup ` calls `/rest/artifact-bundles/backups/archive/{id}` api.
Because into the method `download_backup ` we use `bundle_backup['downloadURI']` that has a `/rest/artifact-bundles/backups/archive/{id}` string.

**Comment by marikrg** [comment]
:+1:
